From b7fd4d3265a1de0d05e2df8f5aa1cdf97879e5ea Mon Sep 17 00:00:00 2001
From: pennam <m.pennasilico@arduino.cc>
Date: Wed, 11 Aug 2021 09:19:53 +0200
Subject: [PATCH] Imgtool: Use 32bit for header size field in image header

---
 scripts/imgtool/image.py | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/scripts/imgtool/image.py b/scripts/imgtool/image.py
index 89edded..d821d06 100644
--- a/scripts/imgtool/image.py
+++ b/scripts/imgtool/image.py
@@ -498,12 +498,11 @@ class Image():
                # type ImageHdr struct {
                'I' +     # Magic    uint32
                'I' +     # LoadAddr uint32
-               'H' +     # HdrSz    uint16
-               'H' +     # PTLVSz   uint16
+               'I' +     # HdrSz    uint32
+               'I' +     # PTLVSz   uint32
                'I' +     # ImgSz    uint32
                'I' +     # Flags    uint32
-               'BBHI' +  # Vers     ImageVersion
-               'I'       # Pad1     uint32
+               'BBHI'    # Vers     ImageVersion
                )  # }
         assert struct.calcsize(fmt) == IMAGE_HEADER_SIZE
         header = struct.pack(fmt,
@@ -516,8 +515,7 @@ class Image():
                 self.version.major,
                 self.version.minor or 0,
                 self.version.revision or 0,
-                self.version.build or 0,
-                0)  # Pad1
+                self.version.build or 0)
         self.payload = bytearray(self.payload)
         self.payload[:len(header)] = header
 
-- 
2.35.1

