From bc65a76a48fe688bf9e9ce3efe90973976cf400a Mon Sep 17 00:00:00 2001
From: pennam <m.pennasilico@arduino.cc>
Date: Mon, 18 Oct 2021 09:41:39 +0200
Subject: [PATCH] Imgtool: Add version placeholder, can be customized during build

---
 scripts/imgtool/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/imgtool/__init__.py b/scripts/imgtool/__init__.py
index ca43b8d..20769a4 100644
--- a/scripts/imgtool/__init__.py
+++ b/scripts/imgtool/__init__.py
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-imgtool_version = "1.8.0"
+imgtool_version = "ARDUINO_VERSION_PLACEHOLDER"
-- 
2.30.2

